### PR TITLE
Fixing printf (and similar) issues

### PR DESCRIPTION
**Pull request by bengadbois**
These were suggested by `go vet`

### CODE REVIEW

File: helper/awsutil/generate_credentials.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (c *CredentialsConfig) GenerateCredentialChain() (*credentials.Credentials,
 				SecretAccessKey: c.SecretKey,
 				SessionToken:    c.SessionToken,
 			}})
-	case c.AccessKey == "" && c.AccessKey == "":
```
Comment (jefferai) [inline helper/awsutil/generate_credentials.go]: This is actually meant to be `case c.AccessKey == "" && c.SecretKey == ""`

File: vault/token_store.go [hunk 2]
```diff
@@ -2172,7 +2172,7 @@ func (ts *TokenStore) tokenStoreRoleCreateUpdate(
 				resp = &logical.Response{}
 			}
 			resp.AddWarning(fmt.Sprintf(
-				"Given explicit max TTL of %d is greater than system/mount allowed value of %d seconds; until this is fixed attempting to create tokens against this role will result in an error",
```
Comment (jefferai) [inline vault/token_store.go]: We actually want this to be an integer, not a float, so it should probably stay `%d` but the parameter be cast to an `int64`.

**Comment by jefferai** [comment]
Just a couple of minor things!

**Comment by bengadbois** [comment]
Thanks for the quick reply, I've squashed them into a new commit!

File: vault/token_store.go [hunk 3]
```diff
@@ -2173,7 +2173,7 @@ func (ts *TokenStore) tokenStoreRoleCreateUpdate(
 			}
 			resp.AddWarning(fmt.Sprintf(
 				"Given explicit max TTL of %d is greater than system/mount allowed value of %d seconds; until this is fixed attempting to create tokens against this role will result in an error",
-				entry.ExplicitMaxTTL.Seconds(), sysView.MaxLeaseTTL().Seconds()))
+				int(entry.ExplicitMaxTTL.Seconds()), sysView.MaxLeaseTTL().Seconds()))
```
Comment (jefferai) [inline vault/token_store.go]: This should be an `int64`, and you only `int64`'d the first parameter -- both should be.

**Comment by jefferai** [comment]
@bengadbois One thing still needs to be updated. BTW, this may not be the case with other projects you've encountered, but the Vault team generally prefers if commits are added to a PR rather than squashed. Squashing makes it difficult to follow the evolution of a PR and makes it impossible to verify changes made as a result of a review by simply looking at the intervening commits.

We can always squash the final product when merging...let us do that part :-)

**Comment by bengadbois** [comment]
Ah okay, sorry about the squashing (I literally was just pushing *another* squashed one as I read this!). I'll make a point to do that in all future PRs. That might also be something to add to CONTRIBUTING.md.

Is this a big enough deal to warrant a new PR, or is this okay?

Also the casting issue should be fixed now.

**Comment by jefferai** [comment]
Nope, no need for a different PR, just an FYI. Thanks for this!

**Comment by jefferai** [comment]
Nope, no need for a different PR, just an FYI. Thanks for this!

**Pull request merged**